### PR TITLE
34 internship page

### LINKED ISSUE #34: ✨ Create SSW Internship Program page
This is priority after the core website infra is done

```[tasklist]
### Tasks
- [ ] Create a new page Implementing the Figma designs linked below
- [ ] Add a link to the page on the MegaMenu under Training | SSW Internship Programs | Fullstack developer
- [ ] Add a link to the new page on About Us | Who we are | Employment Opportunities | Current Positions
```

https://www.figma.com/file/6IU1jg9Stosfecs4YfawoR/SSW-Internship-Program

As per my conversation with @pierssinclairssw the new page url should be ssw.com.au/training/internship/fullstack (confirm this with Adam + Camilla)

### PR DESCRIPTION
**Pull request by BrookJeynes**
Closes: #34 

Added internship page at /training/fullstack

What still **needs to be done**:
- Content needs to be created
- Subscription to mailing list needs the logic written for it to work

Current state:
image
image
image
image

### CODE REVIEW

File: components/blocks/clientLogos.tsx [hunk 1]
```diff
@@ -2,50 +2,46 @@ import React from "react";
...
+    "allianz": "allianz.jpg",
+    "carnival": "carnival.jpg",
+    "cisco": "cisco.jpg",
+    "commonwealthbank": "commbank.jpg",
+    "symantec": "symatec.jpg",
+    "domain": "domain.jpg",
+    "eventcinemas": "events.jpg",
+    "microsoft": "microsoft.jpg",
+    "toll": "toll.jpg",
+};
```
Comment (bradystroud) [inline components/blocks/clientLogos.tsx:15]: Should these be hardcoded?
Can we move them to tina?

File: components/blocks/clientLogos.tsx [hunk 2]
```diff
@@ -2,50 +2,46 @@ import React from "react";
...
-    {
-      type: "string",
-      label: "Alt text",
-      name: "altText",
-      required: true,
-    },
-  ],
+    name: "ClientLogos",
+    label: "Client Logos",
+    // Todo: Find a way to have no fields - the one below is to satisfy compiler
```
Comment (bradystroud) [inline components/blocks/clientLogos.tsx:38]: Investigate this

File: components/training/training.module.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.carouselSubject {
+    width: 830px !important;
+    display: block !important;
+}
+
+/* @media screen and (max-width: 1280px) { */
+/*     .carouselSubject { */
+/*         display: none !important; */
+/*     } */
+/* } */
```
Comment (bradystroud) [inline components/training/training.module.css]: Remove if not used
```suggestion
```

File: content/internship/fullstack.mdx
Comment (bradystroud) [inline content/internship/fullstack.mdx:1]: Whats this for?

File: content/training/fullstack.mdx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (bradystroud) [inline content/training/fullstack.mdx]: 🤔

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: components/blocks/recurringEvent.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+const renderer = ({ days, hours, minutes }) => {
+    return (
+        <div className="flex items-center">
+            <TimeComponent time={days} identifier="days" className="pr-1" />
+            <TimeComponent time={hours} identifier="hrs" className="pr-1" />
+            <TimeComponent time={minutes} identifier="min" className="pr-1" />
+        </div>
+    )
+
+};
```
Comment (Harry-Ross) [inline components/blocks/recurringEvent.tsx]: Maybe change this function name to something clearer - call it some form of countdown component

File: components/training/trainingHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            }
+        </Section>
+    );
+}
+
+export const TrainingCarousel = ({ data }) => {
+    return (
+        <Carousel
+            infiniteLoop
+            // autoPlay
```
Comment (Harry-Ross) [inline components/training/trainingHeader.tsx]: remove

File: components/blocks/mdxComponentRenderer.tsx [hunk 7]
```diff
@@ -11,83 +11,110 @@ import { GoogleMapsWrapper } from "./googleMapsWrapper";
...
+    ClientLogos: Record<string, never>;
+    CustomImage: {
+        src: string;
+        alt: string;
+        height: number;
+        width: number;
+    };
+    RecurringEvent: {
+        applyLinkRedirect: string,
+    };
```
Comment (Harry-Ross) [inline components/blocks/mdxComponentRenderer.tsx]: Should this also take in a day?

File: components/blocks/verticalListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import Image from "next/image";
+import type { Template } from "tinacms";
+import { TinaMarkdown } from "tinacms/dist/rich-text";
+
+export const VerticalListItem = ({ data }) => {
+    const iconScale = data?.iconScale ? data?.iconScale : 1;
```
Comment (Harry-Ross) [inline components/blocks/verticalListItem.tsx]: Can be `const iconScale = data?.iconScale | 1`

File: components/blocks/verticalListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+    return (
+        <div className="flex items-center pb-5">
+            {data.icon &&
+                <Image
+                    src={data.icon || ""}
+                    alt={`${data.title} icon`}
+                    width={65 * iconScale}
+                    height={65}
```
Comment (Harry-Ross) [inline components/blocks/verticalListItem.tsx]: Does it matter if height is multiplied by iconScale

File: components/training/trainingHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+import { Container } from "../util/container";
+import { Section } from "../util/section";
+import { Carousel } from "react-responsive-carousel";
+import styles from "./training.module.css";
+
+const TrainingHeader = ({ data }) => {
+    return (
+        <Section
+            className="flex h-full flex-col items-center border-b-8 border-sswRed bg-white bg-cover bg-no-repeat lg:block"
+            style={{ backgroundImage: `url(${data?.heroBackground})` }}
```
Comment (Harry-Ross) [inline components/training/trainingHeader.tsx]: what happens if its undefined

File: components/training/training.module.css [hunk 11]
```diff
@@ -0,0 +1,5 @@
+.carouselSubject {
+    width: 470px !important;
+    display: block !important;
+}
```
Comment (Harry-Ross) [inline components/training/training.module.css:6]: Add comment to justify CSS module (sin)

File: styles.css [hunk 12]
```diff
@@ -10,6 +10,7 @@
 	html,
 	body {
 		@apply scroll-smooth;
+        font-family: --var(open-sans-font), Helvetica Neue, Helvetica, Arial, sans-serif;
```
Comment (Harry-Ross) [inline styles.css]: @wicksipedia is this fine with the `next/font` additions you made? same with the changes in tailwind.config.js

**Comment by Harry-Ross** [comment]
Approved once my comments are fixed

File: components/blocks/clientLogos.tsx (hunk 2 above)
Comment (wicksipedia) [reply components/blocks/clientLogos.tsx:38]: the schema could be the list of client images

File: tailwind.config.js [hunk 13]
```diff
@@ -259,7 +265,7 @@ module.exports = {
       typography: (theme) => ({
         DEFAULT: {
           css: {
-						maxWidth: 0,
+            maxWidth: 0,
```
Comment (wicksipedia) [inline tailwind.config.js]: can we kill this line?

File: components/blocks/clientLogos.tsx (hunk 1 above)
Comment (bradystroud) [reply components/blocks/clientLogos.tsx:15]: As per our conversation, create a tech debt PBI to deal with this later

Comment (BrookJeynes) [reply components/blocks/clientLogos.tsx:15]: https://github.com/SSWConsulting/SSW.Website-v3/issues/583

File: components/blocks/clientLogos.tsx (hunk 2 above)
Comment (BrookJeynes) [reply components/blocks/clientLogos.tsx:38]: As per my conversation this will be resolved in https://github.com/SSWConsulting/SSW.Website-v3/issues/583

**Review by bradystroud — APPROVED** [review]

**Review by wicksipedia — APPROVED** [review]

**Review by wicksipedia — APPROVED** [review]

**Pull request merged**